third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/WMMA.cpp

```diff
@@ -266,16 +280,31 @@ LogicalResult convertDot(DotOp op, DotOpAdaptor adaptor,
   auto aTensorTy = cast<RankedTensorType>(a.getType());
   auto bTensorTy = cast<RankedTensorType>(b.getType());
   auto dTensorTy = cast<RankedTensorType>(d.getType());
-  auto elemTy = aTensorTy.getElementType();
+  auto aElemTy = aTensorTy.getElementType();
+  auto bElemTy = bTensorTy.getElementType();
+  auto dElemTy = dTensorTy.getElementType();
+
+  const auto kDimOperandSize = aTensorTy.getShape().back();
+
+  std::string intrinsicName;
+  FailureOr<WmmaIntrinsic> maybeWmmaIntrinsic =
+      WmmaIntrinsic::selectFor(wmmaVer, mnkDim[0], mnkDim[1], kDimOperandSize,
+                               aElemTy, bElemTy, dElemTy);
+  if (failed(maybeWmmaIntrinsic)) {
+    llvm::report_fatal_error("No match found in WMMA database\n");
```

Comment:
Can you copy https://github.com/triton-lang/triton/pull/7185 for better error message?